docs/extensions/settings-and-config/email-editor-integration.md

```diff
@@ -307,6 +307,94 @@ add_filter( 'woocommerce_email_editor_register_personalization_tags', 'your_plug
 
 To learn more about personalization tags, please see the [personalization tags documentation](https://github.com/woocommerce/woocommerce/blob/trunk/packages/php/email-editor/docs/personalization-tags.md) in the `woocommerce/email-editor` package.
 
+### Providing custom context for personalization tags
+
+Use the `woocommerce_email_editor_integration_personalizer_context_data` filter to provide custom context data to your personalization tags. This is useful when your extension needs to pass additional data (such as subscription details, loyalty points, or custom order metadata) that your personalization tag callbacks can access.
+
+**Filter details:**
+
+| Property | Value |
+| -------- | ----- |
+| Hook name | `woocommerce_email_editor_integration_personalizer_context_data` |
+| Since | 10.5.0 |
+| Parameters | `$context` (array), `$email` (\WC_Email) |
+| Returns | array |
+
+**Parameters:**
+
+- `$context` *(array)* – The existing context data array. This may already contain data from WooCommerce core or other extensions.
+- `$email` *(\WC_Email)* – The WooCommerce email object being processed. You can use this to access the email ID, recipient, and the object associated with the email (such as an order or customer).
+
+**Return value:**
+
+Return an array of custom context data. This array will be merged with the existing context, making your custom data available to all personalization tag callbacks via the `$context` parameter.
```

Comment:
This should likely be updated to reflect the latest changes, as it no longer merges with the context data but instead filters the existing context data.